### PR TITLE
chore: fuzz GetNodeBootstrapping instead of internal AgentBaker functions

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

To support privatizing internal AgentBaker APIs, we need to only fuzz the intentionally public-facing AgentBaker API: GetNodeBootstrapping

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: fuzz/api/main.go [hunk 1]
```diff
@@ -14,9 +15,15 @@ func Fuzz(data []byte) int {
 		return -1
 	}
 
-	// should not panic
-	baker := agent.InitializeTemplateGenerator()
-	baker.GetNodeBootstrappingCmd(&config)
+	baker, err := agent.NewAgentBaker()
+	if err != nil {
+		return -2
```
Comment (alexeldeib) [inline fuzz/api/main.go:20]: I don't think this actually does anything? it deprioritizes invalid input but only -1/0/1 make a difference

File: fuzz/api/main.go [hunk 2]
```diff
@@ -14,9 +15,15 @@ func Fuzz(data []byte) int {
...
+	}
+
+	_, err = baker.GetNodeBootstrapping(context.Background(), &config)
+	if err != nil {
+		return -3
```
Comment (alexeldeib) [inline fuzz/api/main.go:25]: same as above

File: fuzz/api/main.go (hunk 1 above)
Comment (UtheMan) [inline fuzz/api/main.go:20]: what's the meaning of -2, -3  (and other) return codes? Is this some internal thing for fuzzing lib?

File: fuzz/api/main.go [hunk 3]
```diff
@@ -14,9 +15,15 @@ func Fuzz(data []byte) int {
...
+	}
 
-	return 1
+	return 0
```
Comment (alexeldeib) [inline fuzz/api/main.go:28]: this should remain 1 to prioritize valid input as "interesting"

Comment (cameronmeissner) [reply fuzz/api/main.go:28]: oops

File: fuzz/api/main.go (hunk 2 above)
Comment (cameronmeissner) [reply fuzz/api/main.go:25]: oops

File: fuzz/api/main.go (hunk 1 above)
Comment (cameronmeissner) [reply fuzz/api/main.go:20]: oops

File: fuzz/api/main.go (hunk 2 above)
Comment (alexeldeib) [reply fuzz/api/main.go:25]: https://github.com/dvyukov/go-fuzz#usage as ref

File: fuzz/api/main.go (hunk 1 above)
Comment (cameronmeissner) [reply fuzz/api/main.go]: should actually just be -1/0/1

**Review by alexeldeib — APPROVED** [review]

**Review by alexeldeib — APPROVED** [review]

**Pull request merged**